File: csharp/grains/Grains.cs
public static class Grains
{
    public static ulong Square(int n)
    {
        if (n < 1 || n > 64)
            throw new System.ArgumentOutOfRangeException($"Only defined on [1,64], given {n}.");

        return 1UL << n - 1;
    }

    public static ulong Total() => 18_446_744_073_709_551_615UL;
}

File: c/grains/src/grains.c
#include "grains.h"

uint64_t square(uint8_t index)
{
	if (index < 1 || index > 64)
	{
		return 0;
	}

	// Each of the 64 squares on the board are represented by a bit position in a 64 bit ull.
	return 1ULL << (index - 1);
}

uint64_t total(void)
{
	// All squares is the same as all bits set in a 64 bit ull.
	return 18446744073709551615ULL;
}
